pkg/dispatchcli/cmd/auth.go

```diff
@@ -25,10 +31,56 @@ func multiAuth(writers ...runtime.ClientAuthInfoWriter) runtime.ClientAuthInfoWr
 
 // GetAuthInfoWriter constructor an ClientAuthInfoWriter
 func GetAuthInfoWriter() runtime.ClientAuthInfoWriter {
+	// Check if a jwt token was present, if true, return a Header with Bearer token
+	if token := viper.GetString("DISPATCH_TOKEN"); len(token) != 0 {
+		return apiclient.BearerToken(token)
+	}
+
+	// Check if service-account & sign-key are present, gen/sign JWT token
+	serviceAccount := viper.GetString("SERVICE_ACCOUNT")
+	signKeyPath := viper.GetString("SIGN_KEY_PATH")
+	if len(serviceAccount) != 0 && len(signKeyPath) != 0 {
+		token, err := generateAndSignJWToken(serviceAccount, signKeyPath)
+		if err != nil {
+			fmt.Printf("error generating JWT: %s\n", err.Error())
+		}
+		return apiclient.BearerToken(token)
+	}
+
 	// Oauth2Proxy always expects a cookie header even if the server is setup with SkipAuth. Hence, set a dummy default.
 	cookie := "unset"
 	if dispatchConfig.Cookie != "" {
 		cookie = dispatchConfig.Cookie
 	}
 	return apiclient.APIKeyAuth("cookie", "header", cookie)
 }
+
+// Generate and sign JWT,
+func generateAndSignJWToken(serviceAccount, signKeyPath string) (string, error) {
+
+	// TODO: default JWT expiration 1 hour
+	token := jwt.NewWithClaims(jwt.SigningMethodRS256, jwt.MapClaims{
+		"iss": serviceAccount,
+		"exp": time.Now().Add(time.Duration(1) * time.Hour).Unix(),
```

Comment:
could you make the expiration time a constant?

Reply:
sure, fixed